### PR TITLE
schedule: use sync pool to decrease fit peer new

### LINKED ISSUE #5552: using sync.pool to replace new

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?

Issue Number: Close #5552

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

master：
image

This PR：
image

The memory allocation has decrease 1/3，and the performan improve 1.2X

Code changes

Side effects

Related changes

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — COMMENTED** [review]
would you mind pasting the bench result?

File: server/schedule/placement/fit.go [hunk 1]
```diff
@@ -215,28 +219,26 @@ func replaceFitPeerOpt(srcStoreID uint64, dstStore *core.StoreInfo) fitPeerOpt {
...
 		}
 		peers[i] = peer
 	}
+	return peers
+}
+
+func newFitWorker(stores []*core.StoreInfo, region *core.RegionInfo, rules []*Rule) *fitWorker {
+	peers := newFitPeer(stores, region, region.GetPeers())
 	// Sort peers to keep the match result deterministic.
 	sort.Slice(peers, func(i, j int) bool {
```
Comment (rleungx) [inline server/schedule/placement/fit.go:237]: Will `Replace` rely on the sort?

Comment (bufferflies) [reply server/schedule/placement/fit.go:237]: None, The replaced peer doesn't consider the selected order and the peer state will be same, so it doesn't need to sort.

**Review by rleungx — APPROVED** [review]

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**